engelsystem/src/main/java/info/metadude/android/eventfahrplan/engelsystem/EngelsystemServiceProvider.kt

```diff
@@ -0,0 +1,21 @@
+package info.metadude.android.eventfahrplan.engelsystem
+
+import info.metadude.kotlin.library.engelsystem.ApiModule
+import info.metadude.kotlin.library.engelsystem.EngelsystemService
+import okhttp3.OkHttpClient
+
+/**
+ * Indirection provider to allow replacing the service in test scenarios.
+ * The default implementation is used at runtime.
+ */
+interface EngelsystemServiceProvider {
+
+    companion object {
+        fun getNewInstance() = object : EngelsystemServiceProvider {}
```

Comment:
Is this a good way to provide a default-implementation? Especially because you don't have any not-implemented-methods here. Maybe this should be an `open class`, and `getService` an `open fun`?

Reply:
What exactly do you suggest? I inserted the interface to enable testing.

Reply:
I just find it weird that you init an Interface inside the Interface with its default-functions. IMHO, I think an Interface should not do this, but a class should. This might be overkill, but maybe .EngelsystemServiceProvider should only be an interface with one not-implemented function. And then there is some `open class` (Maybe `DefaultEngelsystemServiceProvider` or so), which contains the default-implementation of the function + Can be initalized.

Reply:
I understand your intention. I would not try to fix it here since `EngelsystemServiceProvider` is only glue code. I believe it is more of weakness in the [engelsystem library](https://github.com/johnjohndoe/engelsystem/blob/master/engelsystem-base/src/main/kotlin/info/metadude/kotlin/library/engelsystem/ApiModule.kt) and should be fixed there.

Reply:
This will be fixed in the next release of the engelsystem library, see: https://github.com/johnjohndoe/engelsystem/commit/a95dacbaf0836a2f245f47330e71ac7565fb3be4.

Reply:
@NWuensche Fixed here: https://github.com/EventFahrplan/EventFahrplan/pull/483